tests/s3/test_async_delete.py

```diff
@@ -0,0 +1,446 @@
+        self.log.info("Step 1: Upload objects")
+        for obj_name, fpath in zip(self.obj_names, self.fixed_size_fpaths):
+            resp = self.s3_test_obj.put_object(self.bucket_name, obj_name, fpath)
+            assert_utils.assert_true(resp[0], resp[1])
+        used_capacity1 = get_cortx_capacity()[-1]
+        self.log.info("CORTX capacity after upload: %s", used_capacity1)
+        self.log.info("Step 2: Overwrite objects with random size")
+        overwrite_resp = []
+        total_overwrite_file_size = 0
+        for obj_name, fpath in zip(self.obj_names, self.upload_paths):
+            create_random_file(fpath, DEL_CFG["random_obj_size"]["min"],
+                               DEL_CFG["random_obj_size"]["max"])
+            resp = self.s3_test_obj.put_object(self.bucket_name, obj_name, fpath)
+            assert_utils.assert_true(resp[0], resp[1])
+            overwrite_resp.append(resp)
+            total_overwrite_file_size += os.path.getsize(fpath)
+        self.log.info("Step 3: Sleep for GC minimum wait time")
+        time.sleep(DEL_CFG["gc_min_wait_time_min"] * 60)
+        self.log.info("Step 4: Verify space reclaimed")
+        used_capacity2 = get_cortx_capacity()[-1]
+        self.log.info("CORTX capacity after GC: %s", used_capacity2)
+        if self.total_fixed_file_size <= total_overwrite_file_size:
+            assert_utils.assert_true(used_capacity1 <= used_capacity2, "Space not reclaimed.")
+        else:
+            assert_utils.assert_true(used_capacity1 > used_capacity2, "Space not reclaimed.")
+        self.log.info("Step 5: Verify object overwrite")
+        for obj_name, fpath, resp in zip(self.obj_names, self.upload_paths, overwrite_resp):
+            check_get_head_object_version(self.s3_test_obj, self.s3_ver_obj,
+                                          bucket_name=self.bucket_name, object_name=obj_name,
+                                          etag=resp[1]["ETag"])
+            download_and_check(self.s3_test_obj, self.bucket_name, obj_name,
+                               file_path=fpath, download_path=self.download_path)
+            check_list_objects(self.s3_test_obj, bucket_name=self.bucket_name,
+                               expected_objects=self.obj_names)
+        self.log.info("ENDED: Test object overwrite with GC in an unversioned bucket")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-44825")
+    def test_object_deletion_versioning_enabled_bucket_44825(self):
+        """
+        Test object deletion with GC in a versioning enabled bucket.
+        """
+        self.log.info("STARTED: Test object deletion with GC in a versioning enabled bucket")
+        versions = {}
+        self.log.info("Step 1: Enable bucket versioning")
+        res = self.s3_ver_obj.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(res[0], res[1])
+        self.log.info("Step 2: Upload objects")
```

Comment:
shall we upload couple of times to have 2 version ids ?

Reply:
There were additional steps in this test that would generate delete markers and then perform some verification on them. That is on hold for now and I have automated only parts of the test that are unblocked. Complete steps are present in TDS for reference, in case it helps